### PR TITLE
Rudimentary port of av1_scan_orders to rust.

### PR DESCRIPTION
**Pull request by tmpcnt**
This is a rudimentary port of av1_scan_orders to rust, meaning it was mostly converting the C code to rust via regex and fixing various errors.
Should advance #650 .

### CODE REVIEW

File: src/scan_order.rs [hunk 1]
```diff
@@ -0,0 +1,3669 @@
...
+  105, 105, 112, 112, 112, 71,  78,  78,  85,  85,  92,  92,  99,  99,  106,
+  106, 113, 113, 120, 120, 120, 79,  86,  86,  93,  93,  100, 100, 107, 107,
+  114, 114, 121, 121, 128, 128, 128, 87,  94,  94,  101, 101, 108, 108, 115,
+  115, 122, 122, 129, 129, 136, 136, 136, 95,  102, 102, 109, 109, 116, 116,
+  123, 123, 130, 130, 137, 137, 144, 144, 144, 103, 110, 110, 117, 117, 124,
+  124, 131, 131, 138, 138, 145, 145, 152, 152, 152, 111, 118, 118, 125, 125,
+  132, 132, 139, 139, 146, 146, 153, 153, 160, 160, 160, 119, 126, 126, 133,
+  133, 140, 140, 147, 147, 154, 154, 161, 161, 168, 168, 168, 127, 134, 134,
+  141, 141, 148, 148, 155, 155, 162, 162, 169, 169, 176, 176, 176, 135, 142,
+  142, 149, 149, 156, 156, 163, 163, 170, 170, 177, 177, 184, 184, 18
```
[the remark this answers is not in the document]
Comment (tmpcnt) [reply src/scan_order.rs]: I used the pre-generated tables from libaom (see `av1/common/scan.c`).

File: src/scan_order.rs [hunk 2]
```diff
@@ -0,0 +1,3669 @@
+#![allow(safe_extern_statics)]
+#![allow(non_upper_case_globals)]
+#![allow(dead_code)]
```
[the remark this answers is not in the document]
Comment (tmpcnt) [reply src/scan_order.rs]: Probably not, they are leftovers from a previous attempt. I will remove them in the next commit.
However, disabling ` #![allow(non_upper_case_globals)] ` would generate compiler warnings for most of the constants defined in the file as I did no renaming.

File: src/scan_order.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (lu-zero) [reply src/scan_order.rs]: Then it should be fine.

**Comment by lu-zero** [comment]
I'd squash-merge it and address the nits later. Is it fine for you @tdaede ?

**Comment by tdaede** [comment]
yeah seems ok

**Pull request merged**